db/migrate/20190724055909_move_to_managed_authenticator.rb

```diff
@@ -0,0 +1,24 @@
+class MoveToManagedAuthenticator < ActiveRecord::Migration[5.2]
+  def change
+    ::PluginStoreRow.where(plugin_name: 'oauth2_basic').each do |record|
```

Comment:
We like to keep migrations clear of any specific ruby code - that way they won't break if anything changes with PluginStoreRow or UserAssociatedAccount in future. It should be possible to do the migration in pure SQL, like

https://github.com/discourse/discourse-steam-login/blob/master/db/migrate/20190510055357_migrate_steam_auth_data.rb

You can even do the JSON parsing using something like 

```
plugin_store_row.value::json->'user_id'
```

For now, let's just migrate the data to UserAssociatedAccount, and leave the old data in PluginStoreRows. Then later we can add a post deploy migration to delete the old data (needs to be post deploy so that authentication doesn't break for the few minutes during deployment).

Reply:
Ah cool. I've made this change :+1: